### PR TITLE
Rewrite test-tree generation in Python

### PR DESCRIPTION
**Pull request by kfogel**
This is essentially a fresh start.  I originally thought we might re-use the list of special characters from testTree.php, but then I realized it was better to just loop over the potentially troublemaking regions of ASCII and get all the special characters that way.

### CODE REVIEW

**Comment by kfogel** [comment]
@mithuna Since I needed to start working with the test data, I did a rewrite (in Python, as I don't know PHP very well).  I'm requesting review partly to get your feedback and partly to give you an example of some Python code that's solving a problem you are already familiar with.

@Fenn-CS Feel free to review too, at your leisure!  But I won't block on review to start working with this, since I need to upload some test data pretty soon.

File: generate-tree.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    # -- which you would think would mean we *want* to include it, but
+    # in practice Python's path-accepting functions don't like NULL
+    # bytes either and it's quite unlikely to be found in a user's
+    # filename... I think?  I dunno.  <shrug>  If we need to test it,
+    # we'll figure out how to get it through to disk below.  For now,
+    # I'm skipping it in the interests of not letting a perfect yak be
+    # the enemy of a perfectly okay yak.
+    for ch in (list(range(1, ord('/')))           # SOH through '.'
+               + list(range(ord(':'), ord('A')))  # ':" through '@'
+               + list(range(ord(':'), ord('A')))  # ':" through '@'
```
Comment (Fenn-CS) [inline generate-tree.py:81]: I am sure why this set of Unicode characters have to be repeated.

**Review by Fenn-CS — APPROVED** [review]
I saw something that looks like a repeated line! Otherwise, the algorithm is sweet and clean :)

**Comment by kfogel** [comment]
Thanks for the catch, @Fenn-CS !  Fixed, rebased, and merging now.

**Comment by mithuna** [comment]
Thanks for including me in this PR. So happy to learn new coding techniques  from experts.

**Comment by kfogel** [comment]
Well, I don't know if I qualify as an expert, but I'm glad it was useful, @mithuna!

**Pull request merged**